src/core/engine/mod.rs

```diff
@@ -82,3 +82,9 @@ impl<IO: EngineIO> Engine<IO> {
 		Ok(())
 	}
 }
+
+impl<IO: EngineIO> Default for Engine<IO> {
+	fn default() -> Self {
+		Engine::new()
+	}
+}
```

Comment:
I haven’t wanted a default implementation for Engine as I've planned for it to take a configuration structure in time, plus default tends to imply to me as something of a more lightweight construction?  I'm torn on this...